### DISCUSSION TITLE
How to exactly generate the 3D structure from the SMILES in batch when the mol's net charge is not zero.

### DISCUSSION THREAD

**Discussion by StefanIsSmart** [rdkit/rdkit#6520] [Q&A]
- The SMILES :
'O=N([O-])C1=C(that CN=C1NCCSCc2ncccc2)Ccexact'

-structures read in by RDKit and add H
![image](https://github.com/rdkit/rdkit/assets/96673065/d791e42b-8760-45d8-900a-5c2d468de1a2)

- I want to generate the xyz file by RDKit, but some error happened:
```
#######
## Code 
#######
compound =  'O=N([O-])C1=C(CN=C1NCCSCc2ncccc2)Cc3ccccc3'
mol_ = Chem.MolFromSmiles(compound)
if mol_ is None:
    mol_ = Chem.MolFromSmiles(compound, sanitize=False)
    mol_.UpdatePropertyCache(strict=False)
    Chem.SanitizeMol(mol_,Chem.SanitizeFlags.SANITIZE_FINDRADICALS|Chem.SanitizeFlags.SANITIZE_KEKULIZE|Chem.SanitizeFlags.SANITIZE_SETAROMATICITY|Chem.SanitizeFlags.SANITIZE_SETCONJUGATION|Chem.SanitizeFlags.SANITIZE_SETHYBRIDIZATION|Chem.SanitizeFlags.SANITIZE_SYMMRINGS,catchErrors=True)
else:
    pass

mol_h=Chem.AddHs(mol_)
AllChem.EmbedMolecule(mol_h)
AllChem.MMFFOptimizeMolecule(mol_h)

######
## Error
######
AtomValenceException: Explicit valence for atom # 1 N, 4, is greater than permitted

```

- So I use Open-Babel to generate the 3D structures, the 3D structure (xyz file) generated by Open-Babel like this:
![image](https://github.com/rdkit/rdkit/assets/96673065/c2111870-ce82-4b27-9620-ac98bd6cce75)

You can find it is not consistent with the SMILES!!

In fact, I have so many SMILES that need to generate the exact 3D structures, which means I can not check them by hand. Is there any way to do that?

**Comment by sunhwan** [comment]
Change SMILES to properly assign the charge.

```
compound =  'O=[N+]([O-])C1=C(CN=C1NCCSCc2ncccc2)Cc3ccccc3'
```
